include/dlaf/permutations/general/impl.h

```diff
@@ -131,6 +131,52 @@ void applyPermutationOnCPU(
   }
 }
 
+#if defined(DLAF_WITH_GPU)
+
+template <class T>
+MatrixLayout getMatrixLayout(const matrix::Distribution& distr,
+                             const std::vector<matrix::Tile<T, Device::GPU>>& tiles) {
+  LocalTileSize tile_sz = distr.localNrTiles();
+  MatrixLayout layout;
+  layout.nb = distr.blockSize().rows();
+  layout.ld = tiles[0].ld();
+  layout.row_offset = (tile_sz.rows() > 1) ? tiles[1].ptr() - tiles[0].ptr() : 0;
+  layout.col_offset = (tile_sz.cols() > 1) ? tiles[to_sizet(tile_sz.rows())].ptr() - tiles[0].ptr() : 0;
+  return layout;
+}
+
+template <class T>
+MatrixLayout getMatrixLayout(
+    const matrix::Distribution& distr,
+    const std::vector<matrix::internal::TileAsyncRwMutexReadOnlyWrapper<T, Device::GPU>>& tiles) {
+  const LocalTileSize tile_sz = distr.localNrTiles();
+  MatrixLayout layout;
+  layout.nb = distr.blockSize().rows();
+  layout.ld = tiles[0].get().ld();
+  layout.row_offset = (tile_sz.rows() > 1) ? tiles[1].get().ptr() - tiles[0].get().ptr() : 0;
+  layout.col_offset =
+      (tile_sz.cols() > 1) ? tiles[to_sizet(tile_sz.rows())].get().ptr() - tiles[0].get().ptr() : 0;
+  return layout;
+}
```

Comment:
Optional for this PR since it's unrelated, but I notice that these could probably be combined into one function, using `unwrap(tiles[...])` to replace the use of `.get` or no `.get` depending on if the vector element is a tile or a tile wrapper. The function would have to be more generic though and simply take a `std::vector<T>` so the signature wouldn't mention `Tile` or `TileAsync...`. Not sure if that's a net improvement or if it makes it worse.